pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -318,3 +343,108 @@ func (r *Client) cacert() []byte {
 	}
 	return nil
 }
+
+func (r Client) Finalize(vms []*planapi.VMStatus) {
+	r.connect()
```

Comment:
@mansam this should operate on it's own copy of the client and will outlive `Run()`, I have tested this with 3 VMs, each with 4 precopies, and some of the precopies were still there after `Run()` finished, and were successfully removed

Reply:
Yeah, I see that you didn't use a pointer receiver here in the function signature which should avoid that particular issue.